core/systems/disks/steam_removable_media_manager.gd

```diff
@@ -0,0 +1,181 @@
+
+	# This should never hit if using our device tree, but  the method is public so
+	# make sure.
+	if udisks2.partition_has_protected_mount(partition):
+		logger.error("Attempted to initialize steam library on device with protected mount. Illegal Operation.")
+		return ERR_UNAUTHORIZED
+
+	var drive := "/dev" + partition.dbus_path.trim_prefix(BLOCK_PREFIX)
+	logger.debug("Intitializing partition as Steam Library: " + drive)
+	var args := [drive]
+	var result = await _execute_in_thread(SRM_INIT_MEDIA, args)
+	logger.debug("Got results:", result[0], result[1])
+
+	var success: bool = result[1] == 0
+	if success:
+		return OK
+	return ERR_SCRIPT_FAILED
+
+
+## Calls the SteamRemovableMedia or SteamOS retrigger-automounts script to
+## restart all the media-mount@ scripts.
+func retrigger_automounts() -> Error:
+	if not retrigger_capable:
+		logger.error("System is not retrigger capable")
+		return ERR_UNAVAILABLE
+
+	logger.debug("Retriggering Steam Automounts")
+	var result = await _execute_in_thread(STEAMOS_RETRIGGER_AUTOMOUNTS)
+	logger.debug("Got results:", result[0], result[1])
+
+	var success: bool = result[1] == 0
+	if success:
+		return OK
+	return ERR_SCRIPT_FAILED
+
+
+## Calls the SteamRemovableMedia or SteamOS format-sd script to format mmcblk0
+## as EXT4 and intialize it as a steam library
+func format_sd_card() -> Error:
+	if not format_sd_capable:
+		logger.error("System is not format capable")
+		return ERR_UNAVAILABLE
+
+	# Make sure the sd card isn't being used as a protected drive.
+	var sd_card: BlockDevice
+	for block in udisks2.get_unprotected_devices():
+		if not block.dbus_path.contains("mmcblk0"):
+			continue
+		sd_card = block
+	if not sd_card:
+		logger.warn("Unable to find SD Card!")
+		return ERR_UNAUTHORIZED
+
+	logger.debug("Formatting SD Card: mmcblk0")
+	var result = await _execute_in_thread(STEAMOS_FORMAT_SDCARD)
+	logger.debug("Got results:", result[0], result[1])
+
+	var success: bool = result[1] == 0
+	if success:
+		return OK
+	return ERR_SCRIPT_FAILED
+
+
+## Calls the SteamRemovableMedia or SteamOS trim-devices script to perform a
+## trim operation on mmcblk0.
+func trim_sd_card() -> Error:
+	if not trim_capable:
+		logger.error("System is not trim capable")
+		return ERR_UNAVAILABLE
+
+	# Make sure the sd card isn't being used as a protected drive.
+	var sd_card: BlockDevice
+	for block in udisks2.get_unprotected_devices():
+		if not block.dbus_path.contains("mmcblk0"):
+			continue
+		sd_card = block
+
+	if not sd_card:
+		logger.warn("Unable to find SD Card!")
+		return ERR_UNAUTHORIZED
+
+	logger.debug("Performing TRIM operation on SD Card: mmcblk0")
+	var result = await _execute_in_thread(STEAMOS_TRIM_DEVICES)
+	logger.debug("Got results:", result[0], result[1])
+
+	var success: bool = result[1] == 0
+	if success:
+		return OK
+	return ERR_SCRIPT_FAILED
+
+
+func _execute_in_thread(path: String, args: Array = []) -> Array:
+	block_operations = true
+	var cmd := Command.new(path, args)
```

Comment:
We may need to modify the `Command` class so we can use the new "skip watchdog" feature you added (as well as use a dedicated thread for this)